### PR TITLE
Add table oci_vault_secret. Closes #337

### LINKED ISSUE #337: Add table oci_vault_secrets
https://registry.terraform.io/providers/hashicorp/oci/latest/docs/data-sources/vault_secrets
https://docs.oracle.com/en-us/iaas/Content/KeyManagement/Concepts/keyoverview.htm

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
arnab@turbotindias-MacBook-Pro oci-test % ./tint.js oci_vault_secret
Test names: [ 'tests/oci_vault_secret' ]
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_vault_secret []

PRETEST: tests/oci_vault_secret
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create
 <= read (data resources)

Terraform will perform the following actions:

  # data.local_file.secret will be read during apply
  # (config refers to values not yet known)
 <= data "local_file" "secret"  {
      + content        = (known after apply)
      + content_base64 = (known after apply)
      + filename       = "/private/var/folders/df/957ty4t146vfy54wm4254kp40000gp/T/tests/oci_vault_secret/terraform/pretest/output.json"
      + id             = (known after apply)
    }

  # null_resource.named_test_resource will be created
  + resource "null_resource" "named_test_resource" {
      + id = (known after apply)
    }

  # oci_kms_key.named_test_resource will be created
  + resource "oci_kms_key" "named_test_resource" {
      + compartment_id       = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
      + current_key_version  = (known after apply)
      + defined_tags         = (known after apply)
      + desired_state        = (known after apply)
      + display_name         = "steampipetest7312"
      + freeform_tags        = (known after apply)
      + id                   = (known after apply)
      + is_primary           = (known after apply)
      + management_endpoint  = (known after apply)
      + protection_mode      = (known after apply)
      + replica_details      = (known after apply)
      + restored_from_key_id = (known after apply)
      + state                = (known after apply)
      + time_created         = (known after apply)
      + time_of_deletion     = (known after apply)
      + vault_id             = (known after apply)

      + key_shape {
          + algorithm = "AES"
          + curve_id  = (known after apply)
          + length    = 16
        }
    }

  # oci_kms_vault.named_test_resource will be created
  + resource "oci_kms_vault" "named_test_resource" {
      + compartment_id         = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
      + crypto_endpoint        = (known after apply)
      + defined_tags           = (known after apply)
      + display_name           = "steampipetest7312"
      + freeform_tags          = (known after apply)
      + id                     = (known after apply)
      + is_primary             = (known after apply)
      + management_endpoint    = (known after apply)
      + replica_details        = (known after apply)
      + restored_from_vault_id = (known after apply)
      + state                  = (known after apply)
      + time_created           = (known after apply)
      + time_of_deletion       = (known after apply)
      + vault_type             = "DEFAULT"
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + lifecycle_state = (known after apply)
  + resource_id     = (known after apply)
  + resource_name   = "steampipetest7312"
  + tenancy_ocid    = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
oci_kms_vault.named_test_resource: Creating...
oci_kms_vault.named_test_resource: Still creating... [10s elapsed]
oci_kms_vault.named_test_resource: Still creating... [20s elapsed]
oci_kms_vault.named_test_resource: Still creating... [30s elapsed]
oci_kms_vault.named_test_resource: Still creating... [40s elapsed]
oci_kms_vault.named_test_resource: Still creating... [50s elapsed]
oci_kms_vault.named_test_resource: Still creating... [1m0s elapsed]
oci_kms_vault.named_test_resource: Still creating... [1m10s elapsed]
oci_kms_vault.named_test_resource: Still creating... [1m20s elapsed]
oci_kms_vault.named_test_resource: Still creating... [1m30s elapsed]
oci_kms_vault.named_test_resource: Still creating... [1m40s elapsed]
oci_kms_vault.named_test_resource: Still creating... [1m50s elapsed]
oci_kms_vault.named_test_resource: Still creating... [2m0s elapsed]
oci_kms_vault.named_test_resource: Still creating... [2m10s elapsed]
oci_kms_vault.named_test_resource: Still creating... [2m20s elapsed]
oci_kms_vault.named_test_resource: Still creating... [2m30s elapsed]
oci_kms_vault.named_test_resource: Still creating... [2m40s elapsed]
oci_kms_vault.named_test_resource: Still creating... [2m50s elapsed]
oci_kms_vault.named_test_resource: Still creating... [3m0s elapsed]
oci_kms_vault.named_test_resource: Still creating... [3m10s elapsed]
oci_kms_vault.named_test_resource: Still creating... [3m20s elapsed]
oci_kms_vault.named_test_resource: Still creating... [3m30s elapsed]
oci_kms_vault.named_test_resource: Still creating... [3m40s elapsed]
oci_kms_vault.named_test_resource: Still creating... [3m50s elapsed]
oci_kms_vault.named_test_resource: Still creating... [4m0s elapsed]
oci_kms_vault.named_test_resource: Still creating... [4m10s elapsed]
oci_kms_vault.named_test_resource: Still creating... [4m20s elapsed]
oci_kms_vault.named_test_resource: Still creating... [4m30s elapsed]
oci_kms_vault.named_test_resource: Still creating... [4m40s elapsed]
oci_kms_vault.named_test_resource: Still creating... [4m50s elapsed]
oci_kms_vault.named_test_resource: Still creating... [5m0s elapsed]
oci_kms_vault.named_test_resource: Still creating... [5m10s elapsed]
oci_kms_vault.named_test_resource: Still creating... [5m20s elapsed]
oci_kms_vault.named_test_resource: Still creating... [5m30s elapsed]
oci_kms_vault.named_test_resource: Still creating... [5m40s elapsed]
oci_kms_vault.named_test_resource: Still creating... [5m50s elapsed]
oci_kms_vault.named_test_resource: Still creating... [6m0s elapsed]
oci_kms_vault.named_test_resource: Still creating... [6m10s elapsed]
oci_kms_vault.named_test_resource: Still creating... [6m20s elapsed]
oci_kms_vault.named_test_resource: Creation complete after 6m27s [id=ocid1.vault.oc1.iad.b5q6k2lnaagn6.abuwcljsgyzbclulalrwon2ydrit6r4wlts22hnevhyx2kjhfvkh6yhfqs6q]
oci_kms_key.named_test_resource: Creating...
oci_kms_key.named_test_resource: Still creating... [10s elapsed]
oci_kms_key.named_test_resource: Still creating... [20s elapsed]
oci_kms_key.named_test_resource: Still creating... [30s elapsed]
oci_kms_key.named_test_resource: Still creating... [40s elapsed]
oci_kms_key.named_test_resource: Still creating... [50s elapsed]
oci_kms_key.named_test_resource: Still creating... [1m0s elapsed]
oci_kms_key.named_test_resource: Still creating... [1m10s elapsed]
oci_kms_key.named_test_resource: Still creating... [1m20s elapsed]
oci_kms_key.named_test_resource: Still creating... [1m30s elapsed]
oci_kms_key.named_test_resource: Still creating... [1m40s elapsed]
oci_kms_key.named_test_resource: Still creating... [1m50s elapsed]
oci_kms_key.named_test_resource: Still creating... [2m0s elapsed]
oci_kms_key.named_test_resource: Still creating... [2m10s elapsed]
oci_kms_key.named_test_resource: Still creating... [2m20s elapsed]
oci_kms_key.named_test_resource: Still creating... [2m30s elapsed]
oci_kms_key.named_test_resource: Still creating... [2m40s elapsed]
oci_kms_key.named_test_resource: Still creating... [2m50s elapsed]
oci_kms_key.named_test_resource: Still creating... [3m0s elapsed]
oci_kms_key.named_test_resource: Still creating... [3m10s elapsed]
oci_kms_key.named_test_resource: Still creating... [3m20s elapsed]
oci_kms_key.named_test_resource: Still creating... [3m30s elapsed]
oci_kms_key.named_test_resource: Still creating... [3m40s elapsed]
oci_kms_key.named_test_resource: Still creating... [3m50s elapsed]
oci_kms_key.named_test_resource: Still creating... [4m0s elapsed]
oci_kms_key.named_test_resource: Still creating... [4m10s elapsed]
oci_kms_key.named_test_resource: Still creating... [4m20s elapsed]
oci_kms_key.named_test_resource: Still creating... [4m30s elapsed]
oci_kms_key.named_test_resource: Still creating... [4m40s elapsed]
oci_kms_key.named_test_resource: Still creating... [4m50s elapsed]
oci_kms_key.named_test_resource: Still creating... [5m0s elapsed]
oci_kms_key.named_test_resource: Still creating... [5m10s elapsed]
oci_kms_key.named_test_resource: Creation complete after 5m16s [id=ocid1.key.oc1.iad.b5q6k2lnaagn6.abuwcljtdbtzidlwefm3kljtbknscij2ktdvdw3leo7ueraqvlm2r3ctbosq]
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "oci vault secret create-base64 --compartment-id ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq --secret-name steampipetest7312 --vault-id ocid1.vault.oc1.iad.b5q6k2lnaagn6.abuwcljsgyzbclulalrwon2ydrit6r4wlts22hnevhyx2kjhfvkh6yhfqs6q --key-id ocid1.key.oc1.iad.b5q6k2lnaagn6.abuwcljtdbtzidlwefm3kljtbknscij2ktdvdw3leo7ueraqvlm2r3ctbosq --secret-content-content VGhpcyBpcyB0ZXN0IHNlY3JldA== --output json > /private/var/folders/df/957ty4t146vfy54wm4254kp40000gp/T/tests/oci_vault_secret/terraform/pretest/output.json"]
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "sleep 60"]
null_resource.named_test_resource: Still creating... [10s elapsed]
null_resource.named_test_resource: Still creating... [20s elapsed]
null_resource.named_test_resource: Still creating... [30s elapsed]
null_resource.named_test_resource: Still creating... [40s elapsed]
null_resource.named_test_resource: Still creating... [50s elapsed]
null_resource.named_test_resource: Still creating... [1m0s elapsed]
null_resource.named_test_resource: Creation complete after 1m4s [id=5283633272889009172]
data.local_file.secret: Reading...
data.local_file.secret: Read complete after 0s [id=d095e9cc70dd73896339d0ca5d704d61c5ec475e]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

lifecycle_state = "CREATING"
resource_id = "ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaayog2gp6z5akzskueacgi6cqvh2e7vmqb4tptmnww6hca"
resource_name = "steampipetest7312"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

TEST: tests/oci_vault_secret
Running terraform

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration
and found no differences, so no changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Running SQL query: test-get-query.sql

[
  {
    "id": "ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaayog2gp6z5akzskueacgi6cqvh2e7vmqb4tptmnww6hca",
    "lifecycle_state": "ACTIVE",
    "name": "steampipetest7312"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "id": "ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaayog2gp6z5akzskueacgi6cqvh2e7vmqb4tptmnww6hca",
    "lifecycle_state": "ACTIVE",
    "name": "steampipetest7312"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "steampipetest7312"
  }
]
✔ PASSED

POSTTEST: tests/oci_vault_secret
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # null_resource.destroy_test_resource will be created
  + resource "null_resource" "destroy_test_resource" {
      + id = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.
null_resource.destroy_test_resource: Creating...
null_resource.destroy_test_resource: Provisioning with 'local-exec'...
null_resource.destroy_test_resource (local-exec): Executing: ["/bin/sh" "-c" "oci vault secret schedule-secret-deletion --secret-id ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaayog2gp6z5akzskueacgi6cqvh2e7vmqb4tptmnww6hca"]
null_resource.destroy_test_resource (local-exec): {
null_resource.destroy_test_resource (local-exec):   "etag": "5ac8c789c894a7c0571556bff7fd0269d67e403505bcdbc7e833743f2a697b6d"
null_resource.destroy_test_resource (local-exec): }
null_resource.destroy_test_resource: Creation complete after 2s [id=4286074098416873378]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

TEARDOWN: tests/oci_vault_secret



SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  id,
  key_id,
  lifecycle_state,
  vault_id
from
  oci_vault_secret;
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------+------------------+--------------------------------------------------------------------------------------------------------+
| name                | id                                                                                             | key_id                                                                                               | lifecycle_state  | vault_id                                                                                               |
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------+------------------+--------------------------------------------------------------------------------------------------------+
| test-secret-hsm-aes | ocid1.vaultsecret.oc1.ap-mumbai-1.amaaaaaa6igdexaagqqu5bts744mfbo6cfs2rf74hyf5qnxhvanrjcojv6dq | ocid1.key.oc1.ap-mumbai-1.cvq6kocbaagno.abrg6ljrwqwsaivesdw4qrgkjhbst3qujw4visfx5iibry7hhqh5h7ad4y5a | ACTIVE           | ocid1.vault.oc1.ap-mumbai-1.cvq6kocbaagno.abrg6ljrc6ntag6kxwhupjbmoue5kl3zkx5jadk4hk7dmdlaayv7t46kaq2a |
| steampipetest2707   | ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaa6ueq7ztldpthktrmotvdrfgpc6g5fo4asvtpqlygr6ka         | ocid1.key.oc1.iad.b5q6kx65aabk6.abuwcljspyb5qpqndvurviv7fiwm2qnmnz3g5qv4ksrzizkx6vlnzfzqjeda         | PENDING_DELETION | ocid1.vault.oc1.iad.b5q6kx65aabk6.abuwcljtv6ntjmahtofmflz4uzozhfqvxb7mqdc4h7uqmrlanj2x7cwluklq         |
| test-secret-hsm-aes | ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaavvjccnu4wwcebkz5cywj7ajkraad6a2mh452t7m5iujq         | ocid1.key.oc1.iad.b5q6ko44aacqe.abuwcljsxjyroykxdmuxtxfowvuw7jwxpkwzzvxjjqdtuwqs2fuehddogmka         | ACTIVE           | ocid1.vault.oc1.iad.b5q6ko44aacqe.abuwcljr7osdpb27o2mnvk3nrfkynurcqc6xebiryghwd46oeoisnf4dw26q         |
| steampipetest7312   | ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaayog2gp6z5akzskueacgi6cqvh2e7vmqb4tptmnww6hca         | ocid1.key.oc1.iad.b5q6k2lnaagn6.abuwcljtdbtzidlwefm3kljtbknscij2ktdvdw3leo7ueraqvlm2r3ctbosq         | PENDING_DELETION | ocid1.vault.oc1.iad.b5q6k2lnaagn6.abuwcljsgyzbclulalrwon2ydrit6r4wlts22hnevhyx2kjhfvkh6yhfqs6q         |
| test-key-hsm-aes    | ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaavau5r462g4xhcnzzb7hynxrzcxt6r75iekwh5sssoukq         | ocid1.key.oc1.iad.b5q6ko44aacqe.abuwcljsxjyroykxdmuxtxfowvuw7jwxpkwzzvxjjqdtuwqs2fuehddogmka         | PENDING_DELETION | ocid1.vault.oc1.iad.b5q6ko44aacqe.abuwcljr7osdpb27o2mnvk3nrfkynurcqc6xebiryghwd46oeoisnf4dw26q         |
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------+------------------+--------------------------------------------------------------------------------------------------------+
> select
  name,
  id,
  lifecycle_state
from
  oci_vault_secret
where
  lifecycle_state = 'PENDING_DELETION';
+-------------------+----------------------------------------------------------------------------------------+------------------+
| name              | id                                                                                     | lifecycle_state  |
+-------------------+----------------------------------------------------------------------------------------+------------------+
| steampipetest7312 | ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaayog2gp6z5akzskueacgi6cqvh2e7vmqb4tptmnww6hca | PENDING_DELETION |
| test-key-hsm-aes  | ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaavau5r462g4xhcnzzb7hynxrzcxt6r75iekwh5sssoukq | PENDING_DELETION |
| steampipetest2707 | ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaa6ueq7ztldpthktrmotvdrfgpc6g5fo4asvtpqlygr6ka | PENDING_DELETION |
+-------------------+----------------------------------------------------------------------------------------+------------------+
> select
  id,
  name,
  jsonb_pretty(secret_rules) as rules
from
  oci_vault_secret;
+------------------------------------------------------------------------------------------------+---------------------+----------------------------------------------------------+
| id                                                                                             | name                | rules                                                    |
+------------------------------------------------------------------------------------------------+---------------------+----------------------------------------------------------+
| ocid1.vaultsecret.oc1.ap-mumbai-1.amaaaaaa6igdexaagqqu5bts744mfbo6cfs2rf74hyf5qnxhvanrjcojv6dq | test-secret-hsm-aes | [                                                        |
|                                                                                                |                     |     {                                                    |
|                                                                                                |                     |         "ruleType": "SECRET_EXPIRY_RULE",                |
|                                                                                                |                     |         "timeOfAbsoluteExpiry": "2022-01-19T16:00:00Z",  |
|                                                                                                |                     |         "secretVersionExpiryInterval": "P2D",            |
|                                                                                                |                     |         "isSecretContentRetrievalBlockedOnExpiry": false |
|                                                                                                |                     |     }                                                    |
|                                                                                                |                     | ]                                                        |
| ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaayog2gp6z5akzskueacgi6cqvh2e7vmqb4tptmnww6hca         | steampipetest7312   | [                                                        |
|                                                                                                |                     | ]                                                        |
| ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaa6ueq7ztldpthktrmotvdrfgpc6g5fo4asvtpqlygr6ka         | steampipetest2707   | [                                                        |
|                                                                                                |                     | ]                                                        |
| ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaavau5r462g4xhcnzzb7hynxrzcxt6r75iekwh5sssoukq         | test-key-hsm-aes    | [                                                        |
|                                                                                                |                     |     {                                                    |
|                                                                                                |                     |         "ruleType": "SECRET_REUSE_RULE",                 |
|                                                                                                |                     |         "isEnforcedOnDeletedSecretVersions": false       |
|                                                                                                |                     |     }                                                    |
|                                                                                                |                     | ]                                                        |
| ocid1.vaultsecret.oc1.iad.amaaaaaa6igdexaavvjccnu4wwcebkz5cywj7ajkraad6a2mh452t7m5iujq         | test-secret-hsm-aes | [                                                        |
|                                                                                                |                     |     {                                                    |
|                                                                                                |                     |         "ruleType": "SECRET_REUSE_RULE",                 |
|                                                                                                |                     |         "isEnforcedOnDeletedSecretVersions": false       |
|                                                                                                |                     |     }                                                    |
|                                                                                                |                     | ]                                                        |
+------------------------------------------------------------------------------------------------+---------------------+----------------------------------------------------------+
> select * from oci_vault_secret where vault_id = 'ocid1.vault.oc1.ap-mumbai-1.cvq6kocbaagno.abrg6ljrc6ntag6kxwhupjbmoue5kl3zkx5jadk4hk7dmdlaayv7t46kaq2a' and lifecycle_state = 'ACTIVE' and  name = 'test-secret-hsm-aes';
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------
| name                | id                                                                                             | key_id                                                                             
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------
| test-secret-hsm-aes | ocid1.vaultsecret.oc1.ap-mumbai-1.amaaaaaa6igdexaagqqu5bts744mfbo6cfs2rf74hyf5qnxhvanrjcojv6dq | ocid1.key.oc1.ap-mumbai-1.cvq6kocbaagno.abrg6ljrwqwsaivesdw4qrgkjhbst3qujw4visfx5ii
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------
> select * from oci_vault_secret limit 1;
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------
| name                | id                                                                                             | key_id                                                                             
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------
| test-secret-hsm-aes | ocid1.vaultsecret.oc1.ap-mumbai-1.amaaaaaa6igdexaagqqu5bts744mfbo6cfs2rf74hyf5qnxhvanrjcojv6dq | ocid1.key.oc1.ap-mumbai-1.cvq6kocbaagno.abrg6ljrwqwsaivesdw4qrgkjhbst3qujw4visfx5ii
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------
> select * from oci_vault_secret where id = 'ocid1.vaultsecret.oc1.ap-mumbai-1.amaaaaaa6igdexaagqqu5bts744mfbo6cfs2rf74hyf5qnxhvanrjcojv6dq';
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------
| name                | id                                                                                             | key_id                                                                             
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------
| test-secret-hsm-aes | ocid1.vaultsecret.oc1.ap-mumbai-1.amaaaaaa6igdexaagqqu5bts744mfbo6cfs2rf74hyf5qnxhvanrjcojv6dq | ocid1.key.oc1.ap-mumbai-1.cvq6kocbaagno.abrg6ljrwqwsaivesdw4qrgkjhbst3qujw4visfx5ii
+---------------------+------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------
```
</details>

### CODE REVIEW

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — CHANGES_REQUESTED** [review]
@c0d3r-arnab 
revisit descriptions as discussed

File: oci/table_oci_vault_secret.go [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+			},
+			{
+				Name:        "vault_id",
+				Description: "The OCID of the Vault in which the secret exists.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "current_version_number",
+				Description: "The version number of the secret version that's currently in use.",
```
Comment (misraved) [inline oci/table_oci_vault_secret.go:80]:
```suggestion
				Description: "The version number of the secret that's currently in use.",
```

File: oci/table_oci_vault_secret.go [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+			},
+			{
+				Name:        "time_of_current_version_expiry",
+				Description: "An optional property indicating when the current secret version will expire.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("TimeOfCurrentVersionExpiry.Time"),
+			},
+			{
+				Name:        "time_of_deletion",
+				Description: "n optional property indicating when to delete the secret.",
```
Comment (misraved) [inline oci/table_oci_vault_secret.go:108]:
```suggestion
				Description: "An optional property indicating when to delete the secret.",
```

**Review by misraved — COMMENTED** [review]
Make changes to the descriptions of the suggested columns

**Review by rajlearner17 — APPROVED** [review]
Looks good after testing required changes.

**Pull request merged**